Automattic/jetpack
# Admin Menu: Use custom menus always in endpoint

Context: p1HpG7-b2s-p2

Fixes https://github.com/Automattic/wp-calypso/issues/49120.

Also related https://github.com/Automattic/wp-calypso/pull/49352

#### Changes proposed in this Pull Request:
This PR ensures the custom admin menus used in the Nav Unification feature are always loaded on requests to the `admin-menu` endpoint, since these requests are originated from contexts where the Nav Unification is enabled (e.g. Calypso).

It also adds some menu items that were missing in Calypso when the Nav Unification is enabled.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

_Jetpack sites_ 
* Go to https://wordpress.com and switch to a Jetpack connected site running this branch.
* Make sure that:
  * My Home is not visible.
  * Plugins links to Calypso.
  * Tools > Marketing is only visible to authors (and higher).
  * Tools > Earn is only visible to admins.
  * Tools > Import links to WP Admin.
  * Tools > Export links to WP Admin.
  * Jetpack > Activity Log is only visible to admins.
  * Jetpack > Backup is only visible to admins.
  * Jetpack > Scan is only visible to admins.
  * Jetpack > Search is always visible.
  * There is WP Admin link at the bottom.

_Atomic sites_
* Go to https://wordpress.com and switch to an Atomic site with Jetpack Beta running this branch.
* Make sure that:
  * My Home is only visible for admins.
  * Plugins links to WP Admin.
  * Tools > Marketing is only visible to authors (and higher).
  * Tools > Earn is only visible to admins.
  * Tools > Import links to Calypso.
  * Tools > Export links to WP Admin.
  * Jetpack > Activity Log is only visible to admins.
  * Jetpack > Backup is only visible to admins.
  * Jetpack > Scan is not visible.
  * Jetpack > Search is always visible.

_Simple sites_
* Apply D56127-code to your WP.com sandbox and sandbox the API.
* Go to https://wordpress.com and switch to a Simple site.
* Make sure that:
  * My Home is only visible for admins.
  * Plugins links to Calypso.
  * Tools > Marketing is only visible to authors (and higher).
  * Tools > Earn is only visible to admins.
  * Tools > Import links to Calypso.
  * Tools > Export links to Calypso.
  * Jetpack > Activity Log is only visible to admins.
  * Jetpack > Backup is only visible to admins.
  * Jetpack > Scan is not visible.
  * Jetpack > Search is always visible.

#### Proposed changelog entry for your changes:
N/A.
